src/bin/grid.rs

```diff
@@ -15,6 +15,19 @@ mod example {
 
     use std::env::args;
 
+    // upgrade weak reference or return
+    #[macro_export]
+    macro_rules! upgrade_weak {
```

Comment:
👍

Reply:
Seems this enough specific macro to be moved to glib

Reply:
It also works on `Arc` and `Rc` btw

Reply:
Indeed, I'll let one of you add it to glib. ;) (and update examples after hehe)